spring-xsuaa-it/src/main/java/testservice/api/multipleBindings/SecurityConfiguration.java

```diff
@@ -0,0 +1,84 @@
+import com.sap.cloud.security.xsuaa.extractor.LocalAuthoritiesExtractor;
+import com.sap.cloud.security.xsuaa.token.TokenAuthenticationConverter;
+import com.sap.cloud.security.xsuaa.token.authentication.XsuaaAudienceValidator;
+import com.sap.cloud.security.xsuaa.token.authentication.XsuaaJwtDecoderBuilder;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+
+@EnableWebSecurity
+@Profile({ "test.api.multiple" })
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		// @formatter:off
+		http.authorizeRequests()
+				.antMatchers("/**").hasAuthority("Display")
+				.anyRequest().denyAll()
+			.and().oauth2ResourceServer()
+			.jwt()
+				.jwtAuthenticationConverter(getJwtAuthenticationConverter());
+		// @formatter:on
+	}
+
+	@Bean
+	@ConfigurationProperties("vcap.services.<<name of your xsuaa instance of plan application>>.credentials")
+	public XsuaaCredentials xsuaaCredentials() {
+		return new XsuaaCredentials(); // primary Xsuaa service binding, e.g. application
+	}
+
+	@Bean
+	public XsuaaServiceConfiguration customXsuaaConfig() {
+		return new XsuaaServiceConfigurationCustom(xsuaaCredentials());
+	}
+
+	@Bean
+	@ConfigurationProperties("vcap.services.<<name of your xsuaa instance of plan broker>>.credentials")
+	public XsuaaCredentials brokerCredentials() {
+		return new XsuaaCredentials(); // secondary Xsuaa service binding, e.g. broker
+	}
+
+	@Bean
+	public JwtDecoder getJwtDecoder() {
+		XsuaaCredentials brokerXsuaaCredentials = brokerCredentials();
+
+		XsuaaAudienceValidator customAudienceValidator = new XsuaaAudienceValidator(customXsuaaConfig());
+		// customAudienceValidator.configureAnotherXsuaaInstance("test3!b1", "sb-clone1!b22|test3!b1");
+		customAudienceValidator.configureAnotherXsuaaInstance(brokerXsuaaCredentials.getXsAppName(), brokerXsuaaCredentials.getClientId());
+		return new XsuaaJwtDecoderBuilder(customXsuaaConfig()).withTokenValidators(customAudienceValidator).build();
+	}
+
+	Converter<Jwt, AbstractAuthenticationToken> getJwtAuthenticationConverter() {
+		LocalAuthoritiesExtractor extractor = new LocalAuthoritiesExtractor(xsuaaCredentials().getXsAppName(), brokerCredentials().getXsAppName());
```

Comment:
can be documented here: 
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/Migration_JavaContainerSecurityProjects.md#multiple-xsuaa-bindings